internal/p2p/queue.go

```diff
@@ -4,6 +4,9 @@ import (
 	tmsync "github.com/tendermint/tendermint/internal/libs/sync"
 )
 
+// default capacity for the size of a queue
+const defaultCapacity uint = 1024 * 1024 * 20 // 20MB
```

Comment:
I'd make this a power of 2 (e.g. 16 or 32) just because that feels right (and also because the statesync messages are typically going to be about this size)

Reply:
This is the closest think to a max chunk size:
```
// chunkMsgSize is the maximum size of a chunkResponseMessage
chunkMsgSize = int(16e6)
```
I'm not too sure how this is enforced (It's used as the `RecvMessageCapacity` which I think is just for the legacy code)

Reply:
Perhaps we should set harder restrictions on the size of the chunk?

Reply:
Yeah I believe `RecvMessageCapacity` is only used in the legacy stack. We don't enforce the `resp` from `LoadSnapshotChunkSync` anywhere in Tendermint.

Reply:
Snapshot chunks? I think for the Cosmos SDK they're hardcoded at 10 MB, but the spec allows up to 16 MB.

Reply:
Ok I think we should enforce this explicitly else if teams use bigger chunks it's not even going to fit on the queue